### PR TITLE
Better result check and tests

### PR DESCRIPTION
**Pull request by aarontp**
This fixes up the results_check() and how it's used to handle invalid objects, and adds some new tests for the run() and run_wrapper() functions.

### CODE REVIEW

File: turbinia/workers/__init__.py [hunk 1]
```diff
@@ -387,6 +398,14 @@ def run_wrapper(self, evidence):
     details from exceptions in the worker cannot be propogated back to the
     Turbinia TaskManager.
 
+    This method should handle (in no particular order):
+      - Exceptions thrown from run()
+      - Verifing valid TurbiniaTaskResult object is returned
+          - Check for bad results (non TurbiniaTaskResults) returned from run()
+          - Auto-close results that haven't been closed
+          - Verifying that the results are serializeable
+      - Locking to make sure only one task is active at a time
```
Comment (berggren) [inline turbinia/workers/__init__.py]: I can't find the locking code here? Or am I missing something? :)

File: turbinia/workers/__init__.py [hunk 2]
```diff
@@ -338,7 +338,7 @@ def touch(self):
     self.last_update = datetime.now()
 
   def result_check(self, result):
```
[the remark this answers is not in the document]
Comment (aarontp) [reply turbinia/workers/__init__.py]: Done.

File: turbinia/workers/__init__.py (hunk 1 above)
Comment (aarontp) [reply turbinia/workers/__init__.py]: Heh, yeah, that's what it *should* do, ;).   I have a separate PR I'm working on for this, but I just added a TODO for it in the meantime.

**Comment by aarontp** [comment]
Conflicts resolved.  PTAL.

**Comment by aarontp** [comment]
Ok, tests are fixed again too (I must have accidentally removed the import line when merging).  PTAL.

**Comment by aarontp** [comment]
Resolved new conflicts.

**Comment by aarontp** [comment]
Ping, :)

**Pull request merged**